lib/super_tooltip.dart

```diff
@@ -1,908 +1,1084 @@
-  /// If a constraint is not set the size will ajust to the content
-  double minWidth, minHeight, maxWidth, maxHeight;
-  ///
-  /// The minium padding from the Tooltip to the screen limits
-  final double minimumOutSidePadding;
-  ///
-  /// If [snapsFarAwayVertically== true] the bigger free space above or below the target will be 
-  /// covered completely by the ToolTip. All other dimension or position constraints get overwritten 
-  final bool snapsFarAwayVertically;
-  ///
-  /// If [snapsFarAwayHorizontally== true] the bigger free space left or right of the target will be 
-  /// covered completely by the ToolTip. All other dimension or position constraints get overwritten 
-  final bool snapsFarAwayHorizontally;
-  /// [top], [right], [bottom], [left] position the Tooltip absolute relative to the whole screen
-  double top, right, bottom, left;
-  ///
-  /// A Tooltip can have none, an inside or an outside close icon
-  final ShowCloseButton showCloseButton;
-  ///
-  /// [hasShadow] defines if the tooltip should have a shadow
-  final bool hasShadow;
-  ///
-  /// the stroke width of the border
-  final double borderWidth;
-  ///
-  /// The corder radii of the border
-  final double borderRadius;
-  ///
-  /// The color of the border
-  final Color borderColor;
-  /// 
-  /// The color of the close icon
-  final Color closeButtonColor;
-  /// 
-  /// The size of the close button 
-  final double closeButtonSize;
-  ///
-  /// The length of the Arrow
-  final double arrowLength;
-  /// 
-  /// The width of the arrow at its base
-  final double arrowBaseWidth;
-  ///
-  /// The distance of the tip of the arrow's tip to the center of the target
-  final double arrowTipDistance;
-  ///
-  /// The backgroundcolor of the Tooltip
-  final Color backgroundColor;
-  /// The color of the rest of the overlay surrounding the Tooltip.
-  /// typically a translucent color.
-  final Color outsideBackgroundColor;
-  ///
-  /// By default touching the surrounding of the Tooltip closes the tooltip.
-  /// you can define a rectangle area where the background is completely transparent 
-  /// and the widgets below react to touch
-  final Rect touchThrougArea; 
-  ///
-  /// The shape of the [touchThrougArea]. 
-  final ClipAreaShape touchThroughAreaShape;
-  ///
-  /// If [touchThroughAreaShape] is [ClipAreaShape.rectangle] you can define a border radius
-  final double touchThroughAreaCornerRadius;
-  ///
-  /// Let's you pass a key to the Tooltips cotainer for UI Testing
-  final Key tooltipContainerKey;
-
-  Offset _targetCenter;
-  OverlayEntry _backGroundOverlay;
-  OverlayEntry _ballonOverlay;
-
-  SuperTooltip({
-    this.tooltipContainerKey,
-    @required this.content, // The contents of the tooltip.
-    @required this.popupDirection,
-    this.onClose,
-    this.minWidth,
-    this.minHeight,
-    this.maxWidth,
-    this.maxHeight,
+  Future<void> hideTooltip() {
+    event = _Event.hide;
+    _completer = Completer();

```

Comment:
I call the completer when i call either `_showTooltip` or `_hideTooltip`.

```dart
_showTooltip() async {
  widget.onShow?.call();

  if (_entry != null) return; // Already visible.

  _createOverlayEntries();

  await _animationController
      .forward()
      .whenComplete(_superTooltipController.complete);
}
```

After the animation controller has finished animating the widget either in or out, calling `_superTooltipController.complete` will toggle the `isVisible` field in the controller. The idea was partially taken from [transformer_page_view](https://github.com/best-flutter/transformer_page_view/blob/master/lib/index_controller.dart) plugin.

Let me know if it's necessary or not. I haven't found a use case but thought it might be expected in the controller.